### ISSUE TITLE
Botocore 1.28 Breaks .save()

### ISSUE THREAD

**Issue by brian-t-otts** [pynamodb/PynamoDB#1092]
In the last week, the project I'm working on migrated from Pipenv to Poetry. On Oct 24th (just a few days ago at the time of this post), AWS released botocore v1.28.0. This is a relatively new project, so I just manually handled the migration from Pipenv to Poetry, so I ignored the Pipfile.lock. After moving to Poetry, pynamodb pulled the latest version of botocore, which was v1.28.3 at the time. At that point, I started receiving the following error when calling .save() on a Model:

```
Traceback (most recent call last):
  File "/var/task/pynamodb/models.py", line 447, in save
    data = self._get_connection().put_item(*args, **kwargs)
  File "/var/task/pynamodb/connection/table.py", line 150, in put_item
    return self.connection.put_item(
  File "/var/task/pynamodb/connection/base.py", line 1011, in put_item
    operation_kwargs = self.get_operation_kwargs(
  File "/var/task/pynamodb/connection/base.py", line 903, in get_operation_kwargs
    operation_kwargs.update(self.get_identifier_map(table_name, hash_key, range_key, key=key))
  File "/var/task/pynamodb/connection/base.py", line 820, in get_identifier_map
    tbl = self.get_meta_table(table_name)
  File "/var/task/pynamodb/connection/base.py", line 548, in get_meta_table
    data = self.dispatch(DESCRIBE_TABLE, operation_kwargs)
  File "/var/task/pynamodb/connection/base.py", line 329, in dispatch
    data = self._make_api_call(operation_name, operation_kwargs, settings)
  File "/var/task/pynamodb/connection/base.py", line 358, in _make_api_call
    request_dict = self.client._convert_to_request_dict(\nTypeError: _convert_to_request_dict() missing 1 required positional argument: 'endpoint_url'"
```

The model is in this format:
```
class MyTable(Model):
    class Meta:
        table_name = os.getenv('TABLE_NAME')
        region = os.getenv('REGION')
    _pk = UnicodeAttribute(hash_key=True)
```

Downgrading to botocore v1.27.96 does fix the issue. Just wanted to let you guys know.

**Comment by ikonst** [comment]
Dup of https://github.com/pynamodb/PynamoDB/issues/1081, addressed in pynamodb 5.2.3.

**Comment by brian-t-otts** [comment]
Thanks for the update. I was working off the 5.2.1, which I must've pulled last week. All's good now. Thanks again!
